src/zenml/integrations/neptune/experiment_trackers/run_state.py

```diff
@@ -0,0 +1,106 @@
+import functools
+import hashlib
+from typing import List
+
+import neptune.new as neptune
+
+from zenml.client import Client
+from zenml.integrations.constants import NEPTUNE
+from zenml.steps.base_step import BaseStepMeta
+
+
+def singleton(class_):
+    instances = {}
+
+    def getinstance(*args, **kwargs):
+        if class_ not in instances:
+            instances[class_] = class_(*args, **kwargs)
+        return instances[class_]
+
+    return getinstance
+
+
+class NoActiveRunException(Exception):
+    pass
+
+
+class InvalidExperimentTrackerSelected(Exception):
+    pass
+
+
+@singleton
+class RunProvider:
+    def __init__(self):
+        self._active_run = None
+        self._project = None
+        self._run_name = None
+        self._token = None
+        self._tags = None
+
+    @property
+    def project(self) -> str:
+        return self._project
+
+    @property
+    def token(self) -> str:
+        return self._token
+
+    @property
+    def run_name(self) -> str:
+        return self._run_name
+
+    @property
+    def tags(self) -> List[str]:
+        return self._tags
+
+    @project.setter
+    def project(self, project: str):
+        self._project = project
+
+    @token.setter
+    def token(self, token: str):
+        self._token = token
+
+    @run_name.setter
+    def run_name(self, run_name: str):
+        self._run_name = run_name
+
+    @tags.setter
+    def tags(self, tags: List[str]):
+        self._tags = tags
+
+    @property
+    def active_run(self) -> neptune.metadata_containers.Run:
+        if self._active_run is None:
+            run = neptune.init_run(
+                project=self.project,
+                api_token=self.token,
+                custom_run_id=hashlib.md5(self.run_name.encode()).hexdigest(),
+                tags=self.tags,
+            )
+
+            self._active_run = run
+        return self._active_run
+
+
+def get_neptune_run() -> neptune.metadata_containers.Run:
+    client = Client()
+    experiment_tracker = client.active_stack.experiment_tracker
+    if experiment_tracker.flavor == NEPTUNE:
+        return experiment_tracker.run_state.active_run
+    raise InvalidExperimentTrackerSelected(
+        "Fetching neptune run works only with neptune flavor of"
+        "experiment tracker selected. Current selection is %s"
+        % experiment_tracker.flavor
+    )
+
+
+def neptune_step(step: BaseStepMeta):
```

Comment:
This is a cute helper function, however it won't work as soon as we enable multiple stack components of a type (e.g. multiple experiment trackers) in a ZenML stack. For this reason I'd suggest getting rid of this and simply doing:
```python
@step(experiment_tracker=Client().active_stack.experiment_tracker.name)
...
```

in the example, as otherwise we'll have issues with backwards compatability related to the `@neptune_step` decorator.

Reply:
Got it!

Reply:
Is there any way how we could make the line more user-friendly?

Reply:
I guess if anything we could make this a little simpler for all stack components by allowing something like
`@step(experiment_tracker=True)` which then uses the experiment tracker configured in the active stack. I can bring this up with the team, but I think for the example I'd suggest doing it as described above.

In general we suggest configuring which experiment tracker is used using a config file anyway instead of defining it in code, but for the examples this is the easiest way to make it work for many users and stacks.

Reply:
@schustmi Do you have any example on how to use a config file to configure usage of an experiment tracker in the pipeline? I cannot find anything like that.

Reply:
The config file always depends on your stack and pipeline, but there's an easy way to generate one:

Instead of calling `my_example_pipeline.run()`, call `my_example_pipeline.write_run_configuration_template("config.yaml")` and you should get a template which you can use.

Reply:
Nice one!